client/jetpack-cloud/components/guided-tour/index.tsx

```diff
@@ -10,50 +11,76 @@ import { preferencesLastFetchedTimestamp } from 'calypso/state/preferences/selec
 import './style.scss';
 
 export interface Tour {
-	targetId: string;
+	target: string;
 	title: string;
 	description: string;
+	popoverPosition?:
+		| 'top'
+		| 'top right'
+		| 'right'
+		| 'bottom right'
+		| 'bottom'
+		| 'bottom left'
+		| 'left'
+		| 'top left';
 }
 
 interface Props {
+	className?: string;
 	tours: Tour[];
 	preferenceName: string;
 }
 
-const GuidedTour = ( { tours, preferenceName }: Props ) => {
+// This hook will return the async element matching the target selector.
+// After timeout has passed, it will return null.
+const useAsyncElement = ( target: string, maxDuration: number ): HTMLElement | null => {
```

Comment:
Not a functional blocker, but `maxDuration` here doesn't give any indication as to what unit of time we're talking about. I assume this is milliseconds, since JavaScript's `Date.now()` returns a number of milliseconds since the Unix epoch, but it's not immediately obvious until I read the implementation.

Reply:
Good point. To eliminate confusion, I added a hook description to explain which unit of time we are referring to here. I also renamed the variable to timeout duration to emphasize which duration we are referring to here.